src/test/java/org/example/auctions/services/AuctionServiceTest.java

```diff
@@ -0,0 +1,163 @@
+package org.example.auctions.services;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class AuctionServiceTest {
+    private AuctionService auctionService;
+
+    @BeforeEach
+    public void beforeEach() {
+        auctionService = new AuctionService();
+    }
+
+    @Test
+    public void verifyCreateAuction() {
+        Auction auction = auctionService.createAuction("Android", type.INC, 500, duration.H2, "Eldor");
+        assertNotNull(auction);
+    }
+
+    @Test
+    public void verifyPublishAuction() {
+        Auction auction = auctionService.createAuction("Android", type.INC, 500, duration.H2, "Eldor");;
+        auctionService.publishAuction(auction.getId());
+        assertEquals(status.PUBLISHED, auction.getStatus());
+    }
+
+    @Test
+    public void verifyDeletedAuction() {
+        Auction auction = auctionService.createAuction("Android", type.INC, 500, duration.H2, "Eldor");
+        auctionService.deleteAuction(auction.getId());
+        assertEquals(status.DELETED, auction.getStatus());
+    }
+
+    @Test
+    public void verifyUpdateAuction() {
+        Auction auction = auctionService.createAuction("Android", type.INC, 500, duration.H2, "Eldor");
+        auctionService.updateAuction(auction.getId(), "Iphone", type.INC, 200, duration.H2);
+        assertEquals("Iphone", auction.getSubject());
+        assertEquals(type.INC, auction.getType());
+        assertEquals(200, auction.getPrice());
+        assertEquals(duration.H2, auction.getDuration());
+    }
+
+    @Test
+    public void validateCreateAuctionThrowsWhenPriceLessZero() {
+        RuntimeException thrown = assertThrows(RuntimeException.class, () -> {
+            Auction auction = auctionService.createAuction("Android", type.INC, -1, duration.H2, "Eldor");
+        });
+        assertEquals("Price is less than zero.", thrown.getMessage());
+    }
+
+    @Test
+    public void validateUpdateAuctionThrowsExceptionWhenAuctionIdNotFound() {
+        RuntimeException thrown = assertThrows(RuntimeException.class, () -> {
+            auctionService.updateAuction("non-existent Id", "Iphone", type.INC, 200, duration.H2);
+        });
+        assertEquals("There is no such Auction Id.", thrown.getMessage());
+    }
+
+    @Test
+    public void validateUpdateAuctionThrowsExceptionWhenAuctionPublished() {
+        RuntimeException thrown = assertThrows(RuntimeException.class, () -> {
+            Auction auction = auctionService.createAuction("Android", type.INC, 500, duration.H2, "Eldor");
```

Comment:
It's better to rewrite it this way:

Auction auction = auctionService.createAuction("Android", type.INC, 500, duration.H2, "Eldor");
auction.publish();
RuntimeException thrown = assertThrows(RuntimeException.class, () -> {
       auctionService.updateAuction(auction.getId(), "Iphone", type.INC, 200, duration.H2);
});
assertEquals("The auction is published, it cannot be updated.", thrown.getMessage());